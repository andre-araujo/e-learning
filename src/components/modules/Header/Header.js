import React from 'react';
import Link from 'next/link';
import { string, node } from 'prop-types';

import {
  Title,
  Container,
  Content,
} from './styles';

function Header({
  title,
  addOn,
}) {
  return (
    <Container>
      <Content>
        <Link href="/">
          <a href="/">
            <Title>
              {title}
            </Title>
          </a>
        </Link>
        {addOn}
      </Content>
    </Container>
  );
}

Header.defaultProps = {
  title: 'E-learning',
  addOn: null,
};

Header.propTypes = {
  title: string,
  addOn: node,
};

export default Header;
