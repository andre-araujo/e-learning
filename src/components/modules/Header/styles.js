import styled from 'styled-components';

import Wrapper from '../../elements/Wrapper';

export const Container = styled.nav`
  background: white;
  border-bottom: 1px solid #ccc;
`;

export const Content = styled(Wrapper)`
  display: flex;
  align-items: center;
  min-height: 70px;
  justify-content: space-between;
`;

export const Title = styled.h1`
  color: #27c2d5;
  font-weight: 600;
  font-size: 20px;
`;
