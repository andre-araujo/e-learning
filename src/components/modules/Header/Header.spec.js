import React from 'react';
import { mount } from 'enzyme';

import Header from './Header';

describe('Header component', () => {
  it('should match snapshot', () => {
    expect(mount(<Header />)).toMatchSnapshot();
  });
});
