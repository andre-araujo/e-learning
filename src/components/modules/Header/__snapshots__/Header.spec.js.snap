// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<styled.nav>
  <Styled(Wrapper)>
    <ForwardRef
      blacklist={
        Array [
          "css",
          "flexWrap",
          "flexDirection",
          "alignItems",
          "justifyContent",
          "css",
          "width",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "fontSize",
          "color",
          "bg",
          "flex",
          "order",
          "alignSelf",
        ]
      }
      flexDirection={
        Array [
          "column",
          "row",
        ]
      }
      is={[Function]}
      pb={25}
      pt={25}
      width="100%"
    >
      <ForwardRef
        blacklist={
          Array [
            "css",
            "width",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "fontSize",
            "color",
            "bg",
            "flex",
            "order",
            "alignSelf",
          ]
        }
        flex={1}
      >
        <Link
          href="/"
        >
          <a
            href="/"
          >
            <styled.h1>
              LearningShelf
            </styled.h1>
          </a>
        </Link>
      </ForwardRef>
    </ForwardRef>
  </Styled(Wrapper)>
</styled.nav>
`;
